text/0000-std-aware-cargo.md

```diff
@@ -0,0 +1,335 @@
+[patch.sysroot]
+core = { path = 'my/local/core' }
+std  = { git = 'https://github.com/example/std' }
+```
+
+> NOTE: The use of `sysroot` as a category may be changed to a less loaded
+> category name. This is likely an area for bikeshedding. `sysroot` will be
+> used for the remainder of the document for consistency.
+
+### Technical Implications
+
+The `patch.sysroot` term will be introduced for patch when referring to components such as `std` and `core`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This RFC introduces new concepts to the use of Rust and Cargo, and could be confusing for current users of Rust who have not had to consider changes to `core` or `std` previously. However, in the normal case, most users are unlikely to need these settings, while they allow users that DO need to make changes to control important steps of the build process.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+> Why is this design the best in the space of possible designs?
+
+This approach borrows from existing behaviors used by Cargo to allow configuration of `core` and `std`, as if they were a regular crate dependency.
+
+This approach also offers an approach that can be developed and applied incrementally, allowing for time to find coner cases not considered by this RFC
+
+> What other designs have been considered and what is the rationale for not choosing them?
+
+To the author of this RFC's knowledge, there are no other open designs, other than the use tools that wrap Cargo entirely, such as [xargo].
+
+[xargo]: https://github.com/japaric/xargo
+
+> What is the impact of not doing this?
+
+By not doing this, Rust will continue to be difficult to use for users and platforms "on the edge", such as new platform developers or embedded and WASM users.
+
+# Prior art
+[prior-art]: #prior-art
+
+* [RFC1133] - This RFC from 2015 proposed making cargo aware of std. I still need to review in more detail to find the parts and syntax that may solve some open questions.
+* [xargo] - This external tool was used to achieve a similar workflow as described above, limited to use with a nightly compiler
+* [Cargo Issue 5002] - This issue proposed a syntax for explicit dependency on std
+* [Cargo Issue 5003] - This issue discussed how to be backwards compatible with crates that don't explicitly mention std
+
+[RFC1133]: https://github.com/rust-lang/rfcs/pull/1133
+[Cargo Issue 5002]: https://github.com/rust-lang/cargo/issues/5002
+[Cargo Issue 5003]: https://github.com/rust-lang/cargo/issues/5003
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+## How are dependencies (or non-dependency) on `core` and `std` specified?
+
+For example in a `no_core` or `no_std` crate, how would we tell Cargo **not** to build the `core` and/or `std` dependencies?
+
+## Should `std` be rebuilt if `core` is rebuilt?
+
+Is it necessary to rebuild `std` using the customized `core`, even if no changes to `std` are necessary?
```

Comment:
Right it's probably better to be explicit about this being like any other create, and implicit on what that means. Otherwise we get a "denormalized" spec that separately specifies identical behavior for stdlib and non-stdlib creates, which is subject to confusion and bit-rot.